### PR TITLE
Add RAII class to lock a surface

### PR DESCRIPTION
**Pull request by DarthGandalf**

### CODE REVIEW

File: sdlpp.hpp [hunk 1]
```diff
@@ -400,6 +400,19 @@ namespace sdl
...
+      public:
+        explicit Lock(Surface& s) : m_s(s) {
+          if (SDL_MUSTLOCK(s.get())) s.lock();
+        }
+        ~Lock() {
+          if (SDL_MUSTLOCK(m_s.get())) m_s.unlock();
+        };
+
+      private:
+        Surface& m_s;
```
Comment (mika314) [inline sdlpp.hpp]: I do not like `m_` prefixes. UwU, can you remove it, plz? 
```suggestion
        Surface& s;
```

**Review by mika314 — CHANGES_REQUESTED** [review]
Thank you for the PR, just one very small request.

**Comment by DarthGandalf** [comment]
Done.

**Comment by DarthGandalf** [comment]
Perhaps same should be added for Texture and Audio

**Pull request merged**